iis-core/src/test/java/eu/dnetlib/iis/core/AbstractOozieWorkflowTestCase.java

```diff
@@ -0,0 +1,233 @@
+			throw new RuntimeException(e);
+		} finally {
+			IOUtils.closeQuietly(stdInput);
+		}
+		
+
+		BufferedReader stdError = new BufferedReader(new
+				InputStreamReader(p.getErrorStream()));
+		try {
+			String s = null;
+			while ((s = stdError.readLine()) != null) {
+				System.out.println(s);
+			}
+			
+		} catch (IOException e) {
+			throw new RuntimeException(e);
+		} finally {
+			IOUtils.closeQuietly(stdError);
+		}
+	}
+	
+	private String chooseProfiles() {
+		return getDeployMode() == DeployMode.LOCAL ? MAVEN_TEST_WORKFLOW_LOCALLY_PROFILE : MAVEN_TEST_WORKFLOW_REMOTELY_PROFILE;
+	}
+	
+	private DeployMode getDeployMode() {
+		String deployModeString  = propertiesReader.getProperty(OOZIE_WORKFLOW_DEPLOY_MODE_KEY);
+		return DeployMode.valueOf(deployModeString);
+	}
+	
+	private String getOozieServiceLoc() {
+		return propertiesReader.getProperty(OOZIE_SERVICE_LOC_KEY);
+	}
+	
+	private String getRunOoozieJobLogFilename() {
+		return propertiesReader.getProperty(OOZIE_RUN_WORKFLOW_LOG_FILE_KEY);
+	}
+	
+	private String getJobId() {
+		
+		String jobId;
+		try {
+			jobId = FileUtils.readFileToString(new File(getRunOoozieJobLogFilename()));
+		} catch (IOException e) {
+			throw new RuntimeException("Unable to read run oozie job log file", e);
+		}
+		Pattern pattern = Pattern.compile("^job: (\\S*)$", Pattern.MULTILINE);
+		Matcher matcher = pattern.matcher(jobId);
+		matcher.find();
+		jobId = matcher.group(1);
+		
+		return jobId;
+	}
+	
+	private Status waitForJobFinish(String jobId, long timeoutInSeconds) {
+		
+		long timeout = 1000L * timeoutInSeconds;
+    	long checkInterval = 1000L * 1;
+    	long startTime = System.currentTimeMillis();
+    	List<Status> jobFinishedStatuses = Lists.newArrayList(Status.SUCCEEDED, Status.FAILED, Status.KILLED, Status.SUSPENDED); 
+		
+    	while ((System.currentTimeMillis()-startTime)<timeout) {
+    		Status status;
+			try {
+				Thread.sleep(checkInterval);
+				status = oozieClient.getJobInfo(jobId).getStatus();
+			} catch (OozieClientException e) {
+				throw new RuntimeException("Unable to check oozie job status", e);
+			} catch (InterruptedException e) {
+				throw new RuntimeException(e);
+			}
+			
+			if (jobFinishedStatuses.contains(status)) {
+				return status;
+			}
+    		
+			System.out.println("Job " + jobId + " is still running with status: " + status + " [" + (System.currentTimeMillis()-startTime) + " ms]");
+//    		log.debug("Job {} is still running with status: {} [{} ms]", new Object[] { jobId, status, System.currentTimeMillis()-startTime });
+    	}
+    	try {
+			log.info(oozieClient.getJobLog(jobId));
+		} catch (OozieClientException e) {
+			log.warn("Unable to check job log");
+		}
+    	Assert.fail("Execution of job " + jobId + " exceeded waiting time limit");
+    	
+    	return null;
+	}
+	
+	private void assertJobStatus(String jobId, Status status) {
+		if (status == Status.FAILED || status == Status.KILLED || status == Status.SUSPENDED) {
+			
+			try {
```

Comment:
What about moving this 'try catch' to a separate private method. You could also use it in waitForJobFinish.